### PR TITLE
WIP: Added sulu recipe

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q             | A
| ------------- | ---
| License       | MIT

This pull-request contains a recipe for "sulu/sulu".

#### Test this PR

```
export SYMFONY_ENDPOINT=https://symfony.sh/r/github.com/symfony/recipes-contrib/142
php -n /usr/local/bin/composer config extra.symfony.allow-contrib true
```

And do the following steps.

#### Example

Add this to composer.json.

```json
{
    ...
    "require": {
        ...
        "jackalope/jackalope-doctrine-dbal": "^1.3",
        "sulu/sulu": "dev-enhancement/explicit-symfony-dependencies",
        "zendframework/zend-stdlib": "^2.3",
        "zendframework/zendsearch": "@dev"
    },
    "require-dev": {
        ...
        "symfony/stopwatch": "^3.4"
    },
    "repositories": [
        {
            "type": "vcs",
            "url": "git@github.com:wachterjohannes/sulu.git"
        }
    ]
}
```

Run `composer update` and ignore the failing `cache:clear`

Change following lines in `bin/console` and `public/index.php`:

```php
$kernel = new Kernel($env, $debug);
```

to:

```php
$kernel = new App\SuluKernel($env, $debug);
```

Add following key to the firewalls in the `config/packages/security.yaml`:

```yaml
security:
    firewalls:
        admin: ~
```

The config will be added to this key in another file but the SecurityBundle won't be able to handle that without this line.

After that run `composer install` to finish the installation.

To initialize the application `bin/console sulu:build dev --destroy`

#### TODO 

- [ ] `sulu.cache_dir` has to be set in the kernel (warmup?) and the config in sulu_core has no effect
- [ ] zend-search
- [x] zend-search data directory
- [x] `sulu:translate:export` do file to `web`
- [ ] `symfony/stopwatch` is hard dependency
- [ ] insternal php server crashes`php -S 127.0.0.1:8000 -t public` because `public/admin` folder exists and he tries to find `public/admin/index.php`
- [ ] MonologBundle is not installed then (dependecy of sulu?)
- [x] `The command defined in "Sulu\Bundle\PreviewBundle\Command\CacheClearCommand" cannot have an empty name.` when calling `server:run`
- [ ] Preview
- [x] Upload image (path)
- [ ] Rendering images
- [ ] Recipe for massive_search and massive_build
- [ ] Refactor SuluKernel to reuse more code from Kernel

#SymfonyConHackday2017

### CODE REVIEW

File: sulu/sulu/1.6/src/SuluKernel.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+use Symfony\Bundle\FrameworkBundle\Kernel\MicroKernelTrait;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Routing\RouteCollectionBuilder;
+
+class SuluKernel extends Kernel
+{
+    /**
+     * @var string string
```
Comment (nicolas-grekas) [inline sulu/sulu/1.6/src/SuluKernel.php]: typo

File: sulu/sulu/1.6/src/SuluKernel.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+     */
+    private $configuration;
+
+    public function __construct(string $environment, $debug)
+    {
+        parent::__construct($environment, $debug);
+
+        $this->context = $this->detectContext();
+
+        $configurations = require dirname(__DIR__) . '/config/contexts.php';
```
Comment (nicolas-grekas) [inline sulu/sulu/1.6/src/SuluKernel.php]: should use getConfigDir() I guess?

Comment (wachterjohannes) [reply sulu/sulu/1.6/src/SuluKernel.php]: correct :) but this is more like a working prototype! we will refactor it if we are finished. i will ping you then

**Comment by Nyholm** [comment]
Any updates here?

**Comment by wachterjohannes** [comment]
@Nyholm there will be soon but we have to wait for 2.0 release.

**Comment by Nyholm** [comment]
Let me know when 2.0 is released.

**Comment by Nyholm** [comment]
Hey @wachterjohannes, 
Is 2.0 released yet?

**Comment by wachterjohannes** [comment]
@Nyholm no we have quite a few problems with flex - and we are currently trying to create a skeleton for sulu (https://github.com/wachterjohannes/sulu-skeleton) - so i will close this PR and maybe reopen it later on

**Pull request closed without merging**